BaseStation/MainWindow/MapDisplay.xaml.cs

```diff
@@ -55,20 +47,26 @@ public MapDisplay()
         {
             InitializeComponent();
             BaseServer.GPSUpdate += UpdateRoverPosition;
-            _roverIconBitmap = new BitmapImage(new Uri(Directory.GetCurrentDirectory() + @"/Icons/RoverIcon.png", UriKind.Absolute));
-            _waypointBitmap = new BitmapImage(new Uri(Directory.GetCurrentDirectory() + @"/Icons/waypoint.png", UriKind.Absolute));
+            _roverIconBitmap = LoadImageFromRelativeFile("Icons/RoverIcon.png");
+            _waypointBitmap = LoadImageFromRelativeFile("Icons/waypoint.png");
             RoverIcon = new Image { Source = _roverIconBitmap };
         }
 
+        private static BitmapImage LoadImageFromRelativeFile(string file)
+        {
+            return new BitmapImage(new Uri(Path.Combine(Directory.GetCurrentDirectory(), file), UriKind.Absolute));
+        }
+
         public void DisplayMap(string mapSetFile)
         {
             ClearCanvas();
             // load in individual images
             // TODO get the path from the settings
-            if (File.Exists(Directory.GetCurrentDirectory() + @"\Images\" + mapSetFile))
+            string path = Path.Combine(Directory.GetCurrentDirectory(), "Images", mapSetFile);
```

Comment:
Crashes here is mapSetFile is null

Reply:
Why would mapSetFile be null? It's provided as a parameter. DisplayMap shouldn't be called if the file is null.

Reply:
In updateMapWaypoints in MainWindow.xaml.cs, DisplayMap is called without regarding the nullity of the mapSetfile